### PR TITLE
Scheduler tests v1

### PR DESCRIPTION
**Pull request by lmars**
I have moved `fakeCluster` into a shared `testutils` package, and added a test for `syncCluster` which checks that the context includes the correct formations after a sync.

### CODE REVIEW

File: scheduler/scheduler_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+func (c *fakeControllerClient) GetRelease(releaseID string) (*ct.Release, error) {
+	return c.releases[releaseID], nil
+}
+
+func (c *fakeControllerClient) GetArtifact(artifactID string) (*ct.Artifact, error) {
+	return c.artifacts[artifactID], nil
+}
+
+func (c *fakeControllerClient) GetFormation(appID, releaseID string) (*ct.Formation, error) {
+	return c.formations[formationKey{appID, releaseID}], nil
```
Comment (titanous) [inline scheduler/scheduler_test.go]: These methods should all return `client.ErrNotFound` if the key is not present in the map.

File: scheduler/scheduler_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		Jobs: []*host.Job{
+			{ID: "job0", Attributes: map[string]string{"flynn-controller.app": appID, "flynn-controller.release": release.ID, "flynn-controller.type": "web"}},
+		},
+	}})
+
+	cx := newContext(cc, cl)
+	cx.syncCluster()
+
+	formation := cx.formations.Get(appID, release.ID)
+	c.Assert(formation.AppID, Equals, appID)
```
Comment (titanous) [inline scheduler/scheduler_test.go:79]: I'd also add a `Len` method to `Formations` and assert that it returns `1`, it will probably be useful later.

**Comment by titanous** [comment]
LGTM pending comments.

**Comment by lmars** [comment]
@titanous updated re comments: 15a1b41

**Pull request merged**